src/ott/geometry/costs.py

```diff
@@ -294,6 +291,16 @@ def h_legendre(self, z: jnp.ndarray) -> float:
     q = jax.lax.stop_gradient(self.prox_reg(z))
     return jnp.sum(q * z) - self.h(q)
 
+  def tree_flatten(self):
+    """Flatten tree."""
+    return (), (self.gamma,)
+
+  @classmethod
+  def tree_unflatten(cls, aux_data, children):
+    """Unflatten tree."""
+    del children
+    return cls(*aux_data)
```

Comment:
I moved these up in the class hierarchy, not sure if desired, tests seem to pass locally.